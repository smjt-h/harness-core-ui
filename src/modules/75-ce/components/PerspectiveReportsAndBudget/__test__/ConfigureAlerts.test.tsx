/*
 * Copyright 2021 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import React from 'react'
import { render } from '@testing-library/react'
import type { Budget } from 'services/ce'
import ConfigureAlerts from '../CreateBudgetSteps/ConfigureAlerts'

import MockBudget from './PerspectiveBudgetsResponse.json'

jest.mock('framework/strings', () => ({
  useStrings: () => ({
    getString: (key: string) => key
  })
}))

describe('Test Cases For ConfigureAlerts', () => {
  test('Should be able to render ConfigureAlerts', () => {
    const { container } = render(
      <ConfigureAlerts
        isEditMode={true}
        viewId="mock_id"
        name="mock_name"
        accountId="mock_account_id"
        budget={MockBudget.data[0] as Budget}
        onSuccess={jest.fn()}
      />
    )

    expect(container).toMatchSnapshot()
  })
})
