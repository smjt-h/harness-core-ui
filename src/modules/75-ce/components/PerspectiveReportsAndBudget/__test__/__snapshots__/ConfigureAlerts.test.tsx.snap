// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Cases For ConfigureAlerts Should be able to render ConfigureAlerts 1`] = `
<div>
  <div
    class="StyledProps--font StyledProps--main"
  >
    <div
      class="OverlaySpinner--overlaySpinner"
    >
      <form
        action="#"
        class="FormikForm--main"
      >
        <div>
          <div
            class="StyledProps--font StyledProps--main selectPerspectiveContainer"
          >
            <div
              class="Text--withDocsTooltip"
              data-tooltip-id="createBudgetConfigureAlerts"
            >
              <p
                class="StyledProps--font StyledProps--main StyledProps--font-variation-h4"
              >
                ce.perspectives.budgets.configureAlerts.title
              </p>
            </div>
            <p
              class="StyledProps--font StyledProps--main StyledProps--margin-top-xlarge StyledProps--margin-bottom-xlarge"
            >
              ce.perspectives.budgets.configureAlerts.subTitle
            </p>
            <div
              class="StyledProps--font StyledProps--main StyledProps--margin-bottom-xlarge"
            >
              <span
                class="StyledProps--font StyledProps--main StyledProps--inline StyledProps--color StyledProps--color-grey800 StyledProps--margin-right-small"
              >
                ce.perspectives.budgets.configureAlerts.budgetAmount
              </span>
              <span
                class="StyledProps--font StyledProps--main StyledProps--inline StyledProps--color StyledProps--color-grey800 StyledProps--font-weight-bold StyledProps--font-size-medium"
              >
                $0.00
              </span>
            </div>
            <div
              class="StyledProps--font StyledProps--main"
            >
              <div
                class="thresholds"
              >
                <div
                  class="thresholdLabel"
                >
                  <p
                    class="StyledProps--font StyledProps--main label"
                  >
                    ce.perspectives.budgets.configureAlerts.basedOn
                  </p>
                  <p
                    class="StyledProps--font StyledProps--main StyledProps--color StyledProps--color-grey0"
                  >
                    ce.perspectives.budgets.configureAlerts.exceeds
                  </p>
                  <p
                    class="StyledProps--font StyledProps--main label"
                  >
                    ce.perspectives.budgets.configureAlerts.percent
                  </p>
                  <p
                    class="StyledProps--font StyledProps--main label"
                  >
                    ce.perspectives.budgets.configureAlerts.sendAlertTo
                  </p>
                </div>
                <div
                  class="threshCtn"
                >
                  <div
                    class="threshold"
                  >
                    <div
                      class="bp3-form-group"
                      data-id="alertThresholds.0.basedOn-0"
                    >
                      <div
                        class="bp3-form-content"
                      >
                        <div
                          class="bp3-popover-wrapper Select--main bp3-fill"
                        >
                          <div
                            class="bp3-popover-target"
                          >
                            <div
                              class="bp3-input-group"
                            >
                              <input
                                autocomplete="off"
                                class="bp3-input"
                                name="alertThresholds.0.basedOn"
                                placeholder="- Select -"
                                style="padding-right: 0px;"
                                type="text"
                                value="ce.perspectives.budgets.configureAlerts.actual"
                              />
                              <span
                                class="bp3-input-action"
                              >
                                <span
                                  class="bp3-icon bp3-icon-chevron-down StyledProps--main StyledProps--padding-small"
                                  icon="chevron-down"
                                >
                                  <svg
                                    data-icon="chevron-down"
                                    height="14"
                                    viewBox="0 0 16 16"
                                    width="14"
                                  >
                                    <desc>
                                      chevron-down
                                    </desc>
                                    <path
                                      d="M12 5c-.28 0-.53.11-.71.29L8 8.59l-3.29-3.3a1.003 1.003 0 00-1.42 1.42l4 4c.18.18.43.29.71.29s.53-.11.71-.29l4-4A1.003 1.003 0 0012 5z"
                                      fill-rule="evenodd"
                                    />
                                  </svg>
                                </span>
                              </span>
                            </div>
                          </div>
                        </div>
                      </div>
                    </div>
                    <p
                      class="StyledProps--font StyledProps--main pushdown7"
                    >
                      exceeds
                    </p>
                    <div
                      class="bp3-form-group"
                      data-id="alertThresholds.0.percentage-1"
                    >
                      <div
                        class="bp3-form-content"
                      >
                        <div
                          class="bp3-input-group"
                        >
                          <input
                            autocomplete="off"
                            class="bp3-input"
                            name="alertThresholds.0.percentage"
                            type="number"
                            value="80"
                          />
                        </div>
                      </div>
                    </div>
                    <div
                      class="StyledProps--font StyledProps--main"
                    >
                      <div
                        class="bp3-input bp3-tag-input tagInput"
                      >
                        <div
                          class="bp3-tag-input-values"
                        >
                          <span
                            class="bp3-tag tag"
                            data-tag-index="0"
                          >
                            <span
                              class="bp3-text-overflow-ellipsis bp3-fill"
                            >
                              shubhanshu.verma@harness.io
                            </span>
                            <button
                              class="bp3-tag-remove"
                              type="button"
                            >
                              <span
                                class="bp3-icon bp3-icon-small-cross"
                                icon="small-cross"
                              >
                                <svg
                                  data-icon="small-cross"
                                  height="16"
                                  viewBox="0 0 16 16"
                                  width="16"
                                >
                                  <desc>
                                    small-cross
                                  </desc>
                                  <path
                                    d="M9.41 8l2.29-2.29c.19-.18.3-.43.3-.71a1.003 1.003 0 00-1.71-.71L8 6.59l-2.29-2.3a1.003 1.003 0 00-1.42 1.42L6.59 8 4.3 10.29c-.19.18-.3.43-.3.71a1.003 1.003 0 001.71.71L8 9.41l2.29 2.29c.18.19.43.3.71.3a1.003 1.003 0 00.71-1.71L9.41 8z"
                                    fill-rule="evenodd"
                                  />
                                </svg>
                              </span>
                            </button>
                          </span>
                          <input
                            class="bp3-input-ghost"
                            value=""
                          />
                        </div>
                      </div>
                      <p
                        class="StyledProps--font StyledProps--main StyledProps--font-variation-form-message-danger"
                      />
                    </div>
                    <span
                      class="bp3-icon bp3-icon-ban-circle StyledProps--main pushdown7 StyledProps--color StyledProps--color-grey200"
                      icon="ban-circle"
                      style="cursor: pointer;"
                    >
                      <svg
                        data-icon="ban-circle"
                        height="18"
                        viewBox="0 0 16 16"
                        width="18"
                      >
                        <desc>
                          ban-circle
                        </desc>
                        <path
                          d="M8 0C3.58 0 0 3.58 0 8s3.58 8 8 8 8-3.58 8-8-3.58-8-8-8zm3 9H5c-.55 0-1-.45-1-1s.45-1 1-1h6c.55 0 1 .45 1 1s-.45 1-1 1z"
                          fill-rule="evenodd"
                        />
                      </svg>
                    </span>
                  </div>
                  <div />
                </div>
                <div
                  class="StyledProps--font StyledProps--main StyledProps--margin-medium"
                >
                  <button
                    class="bp3-button Button--button StyledProps--font StyledProps--main addNewAlertBtn Button--with-current-color Button--without-shadow"
                    type="button"
                  >
                    <span
                      class="bp3-button-text"
                    >
                      ce.perspectives.budgets.configureAlerts.createAlert
                    </span>
                  </button>
                </div>
              </div>
            </div>
            <span
              style="flex-grow: 1;"
            />
            <div
              class="Layout--horizontal Layout--layout-spacing-medium StyledProps--main"
            >
              <button
                class="bp3-button Button--button StyledProps--font StyledProps--main Button--with-current-color Button--withLeftIcon"
                type="button"
              >
                <span
                  class="bp3-icon bp3-icon-chevron-left StyledProps--main StyledProps--padding-right-xsmall"
                  icon="chevron-left"
                >
                  <svg
                    data-icon="chevron-left"
                    height="16"
                    viewBox="0 0 16 16"
                    width="16"
                  >
                    <desc>
                      chevron-left
                    </desc>
                    <path
                      d="M7.41 8l3.29-3.29c.19-.18.3-.43.3-.71a1.003 1.003 0 00-1.71-.71l-4 4C5.11 7.47 5 7.72 5 8c0 .28.11.53.29.71l4 4a1.003 1.003 0 001.42-1.42L7.41 8z"
                      fill-rule="evenodd"
                    />
                  </svg>
                </span>
                <span
                  class="bp3-button-text"
                >
                  previous
                </span>
              </button>
              <button
                class="bp3-button Button--button StyledProps--font StyledProps--main StyledProps--intent-primary Button--with-current-color Button--withRightIcon"
                type="button"
              >
                <span
                  class="bp3-button-text"
                >
                  save
                </span>
                <span
                  class="bp3-icon bp3-icon-chevron-right StyledProps--main"
                  icon="chevron-right"
                >
                  <svg
                    data-icon="chevron-right"
                    height="16"
                    viewBox="0 0 16 16"
                    width="16"
                  >
                    <desc>
                      chevron-right
                    </desc>
                    <path
                      d="M10.71 7.29l-4-4a1.003 1.003 0 00-1.42 1.42L8.59 8 5.3 11.29c-.19.18-.3.43-.3.71a1.003 1.003 0 001.71.71l4-4c.18-.18.29-.43.29-.71 0-.28-.11-.53-.29-.71z"
                      fill-rule="evenodd"
                    />
                  </svg>
                </span>
              </button>
            </div>
          </div>
        </div>
      </form>
    </div>
  </div>
</div>
`;
